node/core/pvf/common/src/worker/mod.rs

```diff
@@ -237,9 +240,18 @@ pub mod thread {
 		F: Send + 'static + panic::UnwindSafe,
 		R: Send + 'static,
 	{
-		thread::Builder::new()
-			.name(name.into())
-			.spawn(move || cond_notify_on_done(f, cond, outcome))
+		thread::Builder::new().name(name.into()).spawn(move || {
+			cond_notify_on_done(
+				|| {
+					#[cfg(target_os = "linux")]
+					let _ = crate::worker::security::landlock::try_restrict_thread();
```

Comment:
Same for non Linux - a warning that the landlock sandboxing is not available would be a good idea, I think.

Reply:
> I can emit a warning (once on host startup I think)

Hmm, so given the above, I'm not sure how actionable a warning would be. It could just be noise for most operators.

Reply:
yep something like this ... and indeed we should not keep emitting this warning on each validation attempt. Once on startup would be good and sufficient.

Reply:
Just my2cents, people usually don't look at warnings for various reasons, like  their service is started as a demon or there are plenty of other logs, so an warning will be easy to miss. They won't look at the logs unless something critical happens, like the service not starting or getting compromised.

So, if we consider this the recommended way of running the validator then we should refuse to start if the feature is not present and build in an explicit cmdline argument to bypass this check. So, in this way we would prevent people accidentally run the unsecured version and when they do it they have been warned and explicitly chosed to ignore the warning.

Reply:
Just my2cents, people usually don't look at warnings for various reasons, like  their service is started as a demon or there are plenty of other logs, so an warning will be easy to miss. They won't look at the logs unless something critical happens, like the service not starting or getting compromised.

So, if we consider this the recommended way of running the validator then we should refuse to start if the feature is not present and build in an explicit cmdline argument to bypass this check. So, in this way we would prevent people accidentally run the unsecured version and when they do it they have been warned and explicitly chosed to ignore the warning.

Reply:
> So, if we consider this the recommended way of running the validator then we should refuse to start if the feature is not present and build in an explicit cmdline argument to bypass this check.

That's a good idea, I've added it to https://github.com/paritytech/polkadot/pull/7073 (see description). I don't think we can mandate it right now because like half of validators are on too-old kernel versions (according to the visible [telemetry data](https://telemetry.polkadot.io/#stats/0x91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3)). We already need to announce that "secure mode" change in advance, maybe we can also mention that validators should upgrade to kernel 5.13+ to avoid running in "insecure mode". I will bug Will again to see if we can get that announcement going.

Reply:
Indeed this makes sense, I would nevertheless suggest a soft launch:

1. Enable the feature and print warning on startup if not available, hinting that this feature will be mandatory on some future release XY and that the operator should upgrade the machine. In addition this should obviously be part of the release notes as they are less likely to be ignored.
2. Release version XY which makes it mandatory, exiting with a warning that the check can be bypassed by that command line flag.

[Updating the wiki] would be good as well, but is least likely to be read by already operating operators.

Reply:
@eskimor I missed your comment before merging. Here's what we have now, let me know what needs a follow-up:

- [x] 1. Landlock sandboxing enabled, warning printed if not available
- [ ] 1. Hint that feature will be mandatory in a future release (maybe can do this at the same time as https://github.com/w3f/polkadot-wiki/issues/4881? We should have version XY at that point.)
- [x] 1. Include in release notes (this PR has the proper label)
- [ ] 2. Make it mandatory -- will do in https://github.com/paritytech/polkadot/pull/7073